libcamera-sys/c_api/transform.cpp

```diff
@@ -0,0 +1,89 @@
+#include "transform.h"
+#include <cstring>
+
+extern "C" {
+
+static inline libcamera::Transform to_cpp(libcamera_transform_t t) {
+    return *reinterpret_cast<libcamera::Transform *>(&t);
```

Comment:
```
warning: libcamera-sys@0.6.0: c_api/transform.cpp: In function 'libcamera::Transform to_cpp(libcamera_transform_t)':
warning: libcamera-sys@0.6.0: c_api/transform.cpp:7:13: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
warning: libcamera-sys@0.6.0:     7 |     return *reinterpret_cast<libcamera::Transform *>(&t);
warning: libcamera-sys@0.6.0:       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I believe the correct way would be to `memcpy` here instead of `reinterpret_cast`. C++ is weird.